cmd/controller/main.go

```diff
@@ -43,6 +47,27 @@ func main() {
 	lo.Must0(op.AddHealthzCheck("cloud-provider", awsCloudProvider.LivenessProbe))
 	cloudProvider := metrics.Decorate(awsCloudProvider)
 
+	client := op.Manager.GetAPIReader()
+	if client == nil {
+		panic("panic: failed to initialize read-client on startup")
+	}
+	nodeClassList := v1beta1.EC2NodeClassList{}
+	err := client.List(ctx, &nodeClassList)
+	if err != nil {
+		panic(fmt.Sprintf("panic: failed to list ec2nodeclasses on startup, %s", err.Error()))
```

Comment:
I question whether these should be `logging.FromContext().Fatal` vs. `panic`. Panics are typically reserved for programming errors that we don't expect to happen. These are more like startup conditions aren't met, which feels more semantically correct to "fatal" out on